src/rpc/api/modules/validator.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Attestation, AttestationData, BeaconBlock, bytes32, Deposit, Shard, Slot, Eth1Data, uint64,
+  Fork, SyncingStatus, ValidatorDuty, bytes48, bytes, IndexedAttestation
+} from "../../../types";
+import {DB} from "../../../db";
+import {BeaconChain} from "../../../chain";
+import {OpPool} from "../../../opPool";
+
+import {IValidatorApi} from "../interfaces";
+
+export class ValidatorApi implements IValidatorApi {
+  private chain: BeaconChain;
+  private db: DB;
+  private opPool: OpPool;
+
+  public constructor(opts, {chain, db, opPool}) {
+    this.chain = chain;
+    this.db = db;
+    this.opPool = opPool;
+  }
+
+  public async getClientVersion(): Promise<bytes32> {
+    return Buffer.alloc(32);
+  }
+
+  public async getFork(): Promise<{fork: Fork; chainId: uint64}> {
+    // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+    return {} as {fork: Fork; chainId: uint64};
+  }
+
+  public async getGenesisTime(): Promise<uint64> {
+    // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+    return {} as uint64;
+  }
+
+  public async getSyncingStatus(): Promise<boolean | SyncingStatus> {
+    // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+    return {} as boolean | SyncingStatus;
+  }
+
+  public async getDuties(validatorPubkeys: bytes48[]): Promise<{currentVersion: Fork; validatorDuties: ValidatorDuty[]}> {
+    // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+    return {} as {currentVersion: Fork; validatorDuties: ValidatorDuty[]};
+  }
+
+  public async produceBlock(slot: Slot, randaoReveal: bytes): Promise<BeaconBlock> {
+    // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+    return {} as BeaconBlock;
+  }
+
+  public async produceAttestation(slot: Slot, shard: Shard): Promise<IndexedAttestation> {
```

Comment:
I think we want to return an `AttestationData`, see getAttestationData of the BeaconApi

Reply:
Are you sure? `IndexedAttestation` contains `AttestationData`

Reply:
Ok, now I see the issue, `opPool.receiveAttestaion` looks for `AttestationData`

Reply:
I'm looking at https://github.com/ethereum/eth2.0-specs/blob/dev/specs/validator/0_beacon-chain-validator.md#attestations-1

It seems like the validator is supposed to get an AttestationData and turn it into an Attestation, and then publish that.

Reply:
hmmm ok I see what you are saying -- So the beacon chain should send the validator `attestationData` and then the validator turns it into an `IndexedAttestation`
?